lms/djangoapps/courseware/tests/test_views.py

```diff
@@ -258,7 +258,7 @@ def test_jump_to_for_learner_with_staff_only_content(self, is_staff_user, positi
         )
         expected_url += "?{}".format(urlencode({'activate_block_id': six.text_type(staff_only_vertical.location)}))
 
-        assert expected_url == get_legacy_courseware_url(course_key, usage_key, request)
```

Comment:
I dropped the `course_key` param from `get_legacy_courseware_url` because it was completely ignored, and is redundant in the presence of `usage_key`.

Reply:
You're probably aware, but just in case, and as a warning for testing purposes: Some parts of our code extract the `course_key` from the `usage_key`. For Split Mongo courses, this just works by default. For Old Mongo courses, this relies on someone at some point running `map_into_course` (or a `replace` on the `course_key`), to force a course run key onto old mongo usage keys (e.g. `usage_key = UsageKey.from_string(usage_id).map_into_course(course_key)`). If it's not being referenced at all, then sure, it's safe to remove. But it might indicate that at some point it was running that substitution inside the method and was then shifted to have a mapped usage key sent in, which may be significant if you're later changing how it's being called as well.

The `jump_to` method does this substitution, so I think you're fine. I just wanted to flag it in case it factored into other situations down the road.

Reply:
Got it. Thanks for flagging that.